sklearn/tree/_criterion.pyx

```diff
@@ -409,79 +396,66 @@ cdef class ClassificationCriterion(Criterion):
             child to the left child.
         """
         cdef DOUBLE_t* y = self.y
-        cdef SIZE_t y_stride = self.y_stride
-        cdef DOUBLE_t* sample_weight = self.sample_weight
-
-        cdef SIZE_t* samples = self.samples
         cdef SIZE_t pos = self.pos
         cdef SIZE_t end = self.end
 
-        cdef SIZE_t n_outputs = self.n_outputs
-        cdef SIZE_t* n_classes = self.n_classes
-        cdef SIZE_t label_count_stride = self.label_count_stride
-        cdef double* label_count_left = self.label_count_left
-        cdef double* label_count_right = self.label_count_right
-        cdef double* label_count_total = self.label_count_total
```

Comment:
Those line allowed to avoid one pointer access (this->array->elem). This was a small optimisation.